app/migration/parity_check/dynamic_request_content.rb

```diff
@@ -17,14 +17,23 @@ def fetch(identifier)
     # Path ID methods
 
     def statement_id
-      Statement
-        .output_fee
-        .joins(:active_lead_provider)
-        .where(active_lead_provider: lead_provider.active_lead_providers)
-        .order("RANDOM()")
+      Statements::Query.new(lead_provider:)
+        .statements
+        .distinct(false)
+        .reorder("RANDOM()")
         .pick(:api_id)
     end
 
+    def school_id
+      contract_period_id = ContractPeriod.last
```

Comment:
Will this only ever test the parity check for last contract period? maybe getting a random contract period instead?